internal/alias/interceptor.go

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package alias
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/sdk/pbs/controller/protooptions"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+	"google.golang.org/protobuf/types/descriptorpb"
+)
+
+func init() {
+	// register all proto fields that are aliasable.
+	protoregistry.GlobalTypes.RangeMessages(func(m protoreflect.MessageType) bool {
+		registerAliasableFields(m.Descriptor())
+		return true
+	})
+}
+
+type registrationInfo struct {
+	unaliableWithFields []string
+}
+
+// globalAliasableRegistry is a map of proto field's FullName to registrationInfo.
+var globalAliasableRegistry sync.Map
+
+func registerAliasableFields(d protoreflect.MessageDescriptor) {
+	fields := d.Fields()
+	for i := 0; i < fields.Len(); i++ {
+		f := fields.Get(i)
+		name := f.FullName()
+		opts, ok := f.Options().(*descriptorpb.FieldOptions)
+		if !ok {
+			continue
+		}
+		aliasableInfo := proto.GetExtension(opts, protooptions.E_Aliasable).(*protooptions.AliasInfo)
+		if aliasableInfo != nil {
+			ri := &registrationInfo{unaliableWithFields: aliasableInfo.GetUnlessSet().GetFields()}
+			globalAliasableRegistry.Store(name, ri)
+		}
+	}
+}
+
+// aliasLookup is an interface for looking up an alias by its value.
+type aliasLookup interface {
+	lookupAliasByValue(ctx context.Context, value string) (*Alias, error)
+}
+
+// ResolveAliasFields transforms the request by replacing alias values with their
+// corresponding destination ids. If no alias is found or the alias has no
+// destination id, an error is returned.
+func ResolveAliasFields(ctx context.Context, req proto.Message, lookup aliasLookup) (context.Context, error) {
+	const op = "alias.TransformRequest"
```

Comment:
The `op` does not match the function name. Also, you might consider a more self-descriptive function name, perhaps something like `alias.ResolveRequestIds`? Lastly, can you add some more details to the godoc for this function, specifically can you describe how the new proto options that can be set on `req` can impact the behavior of this?